src/main/java/uk/nhs/digital/nhsconnect/nhais/mesh/MongoScheduler.java

```diff
@@ -0,0 +1,67 @@
+package uk.nhs.digital.nhsconnect.nhais.mesh;
+
+import static org.springframework.data.mongodb.core.query.Criteria.where;
+import static org.springframework.data.mongodb.core.query.Query.query;
+
+import java.time.LocalDateTime;
+
+import org.bson.Document;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoOperations;
+import org.springframework.data.mongodb.core.query.Update;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import com.mongodb.client.result.UpdateResult;
+
+@Component
+public class MongoScheduler {
+
+    @Autowired
+    private MongoOperations mongoOperations;
+
+    @Autowired
+    private MeshClient meshClient;
+
+    private static final String MESH_TIMESTAMP = "mesh_timestamp";
+    private static final String TIMESTAMP = "timestamp";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MongoScheduler.class);
+
+    @Scheduled(fixedRate = 60000)
+    public void updateConditionally() {
+        LOGGER.info("Scheduled job for mesh messages fetching started");
+        if (checkTheTimestampUpdatability()) {
+            LOGGER.info("Mesh messages fetching started");
+            for (String messageId : meshClient.getInboxMessageIds()) {
+                meshClient.getMessage(messageId);
+            }
+        } else {
+            LOGGER.info("Mesh messages fetching is postponed: another application instance is fetching now");
+        }
+    }
+
+    private boolean checkTheTimestampUpdatability() {
+        if (mongoOperations.collectionExists(MESH_TIMESTAMP) && mongoOperations.getCollection(MESH_TIMESTAMP).countDocuments() != 0) {
+            UpdateResult result = mongoOperations.updateFirst(query(where(TIMESTAMP).lt(LocalDateTime.now().minusMinutes(5))),
```

Comment:
I'm no mongo expert, but what if we have 2 documents in that collection (else from this method might be executed concurrently)? Will it always update the same one?

Reply:
In theory there should be only 1 row. Even if there are more than one due to a bug in this whole function, it will update both of them and we'll be good. There will be just multiple unnecessary documents in the db

Reply:
`UpdateFirst` means that it won't update more than one I suppose.